releasenotes/notes/restrict-istio-cni-node-daemonset.yaml

```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+releaseNotes:
+  - |
+    **Added** configurable node affinity to istio-cni values.yaml. Can be used to allow excluding istio-cni from being scheduled on specific nodes.
```

Comment:
What are the scenarios of why someone would want to do it?

Reply:
What is the experience with ambient? If you label a namespace but it lands on a pod that don't have istio-cni....

Reply:
Same applies to sidecars

Reply:
The main use case is multi-tenancy. When users have big k8s clusters, they may want to install service mesh only for a subset of services, because:
1. Service mesh fits only for a minor part of their workloads.
2. Some of their workloads require special care and run on dedicated nodes that are not part of service mesh.
3. They deploy production and test workloads in the same cluster and use node selectors to manage these environments.

Reply:
> What is the experience with ambient?

I guess we will need a similar or the same option in ztunnel daemon set. Do you expect it to be implemented while it's still in experimental phase?

> Same applies to sidecars

This is optional setting, so I assume that users should use it consciously. Do you expect a mechanism to prevent injecting sidecars to workloads on nodes without CNI pods? In such a case, injecting would fail silently, as it would not execute at all. In case of missing CNI node, injected sidecar should fail at startup, and this seems better to me.